pkg/client/outofband/client.go

```diff
@@ -34,7 +34,7 @@ type Provider interface {
 	// DidDocServiceFunc returns a function that returns a DID doc `service` entry.
 	// Used when no service entries are specified when creating messages.
 	DidDocServiceFunc() func() (*did.Service, error)
-	ConnRecorder() ConnectionRecorder
+	ConnectionRecorder() (ConnectionRecorder, error)
```

Comment:
Should be storage provider here.

Reply:
Here is a reference - https://github.com/hyperledger/aries-framework-go/blob/0fb64f2818eefefbddc4ce231642488c14362b63/pkg/client/didexchange/client.go#L91

Reply:
@rolsonquadras @sudeshrshetty @troyronda perhaps I did not explain myself correctly. I opened #1477 to explain my views.